java/src/main/java/com/google/cloud/dataproc/templates/kafka/KafkaReader.java

```diff
@@ -62,7 +62,13 @@ public Dataset<Row> readKafkaTopic(SparkSession spark, Properties prop) {
 
     // check on memory constraints
 
+    return getDatasetByMessageFormat(inputData, prop);
+  }
+
+  public Dataset<Row> getDatasetByMessageFormat(Dataset<Row> inputData, Properties prop) {
     Dataset<Row> processedData = null;
+    String kakfaMessageFormat = prop.getProperty(KAFKA_MESSAGE_FORMAT);
+    String kafkaSchemaUrl = prop.getProperty(KAFKA_SCHEMA_URL);
```

Comment:
Since, we are already fetching `kakfaMessageFormat` &  `kafkaSchemaUrl` above from the `Properties prop`. Let us make these as arguments rather and replace `Properties prop` with `String kakfaMessageFormat, String kafkaSchemaUrl` in the function defination.

Reply:
getDatasetByMessageFormat method is being called from the other places like KafkaToGCSDstream hence it is taking Properties as parameter hence no change needed here